odoo/addons/base/models/ir_model.py

```diff
@@ -1518,13 +1518,7 @@ class IrModelConstraint(models.Model):
          'Constraints with the same name are unique per module.'),
     ]
 
-    def _module_data_uninstall(self):
-        """
-        Delete PostgreSQL foreign keys and constraints tracked by this model.
-        """
-        if not self.env.is_system():
-            raise AccessError(_('Administrator access is required to uninstall a module'))
-
+    def unlink(self):
```

Comment:
I suggest to check access rights before actually removing SQL constraints.
```suggestion
    def unlink(self):
        self.check_access_rights('unlink')
        self.check_access_rule('unlink')
```